core/src/main/java/co/aikar/commands/CommandManager.java

```diff
@@ -555,6 +557,69 @@ void injectDependencies(BaseCommand baseCommand) {
         } while (!clazz.equals(BaseCommand.class));
     }
 
+    /**
+     * Instantiates a class with dependencies then registers a command with ACF
+     *
+     * @param commandClass the class of the command
+     */
+    public void registerCommand(Class<? extends BaseCommand> commandClass) {
+        registerCommand(constructCommand(commandClass));
+    }
+
+    /**
+     * Attempts to construct an instance of a command by injecting dependencies registered with
+     * {@link CommandManager#registerDependency(Class, Object)} into either the only constructor or the constructor
+     * annotated with {@link Dependency}.
+     *
+     * @param commandClass the class to instantiate
+     * @return the instance of the command
+     */
+    BaseCommand constructCommand(Class<? extends BaseCommand> commandClass) {
+        Constructor<?> constructor = null;
+        for (Constructor<?> possibleConstructor : commandClass.getConstructors()) {
+            if (annotations.hasAnnotation(possibleConstructor, Dependency.class)) {
+                if (constructor != null) {
+                    throw new IllegalStateException("Command " + commandClass.getName() + " has more than one " +
+                            "constructor annotated @Depenedency.");
+                }
+                constructor = possibleConstructor;
+            }
+        }
+        if (commandClass.getConstructors().length == 1) {
+            constructor = commandClass.getConstructors()[0];
```

Comment:
This appears really cheap. But strikes me as odd that this takes place after searching for annotated constructors.
Would it make more sense to have this first, with the Annotation check as an else to this? Again, so cheap that it doesn't really matter just looks strange.

Everything else looks good to me though - from just looking at it. Appears eligible for merging soon.